### ISSUE TITLE
google_checks.xml not allowing eol left curly for switch statement with lambda-like construct

### ISSUE THREAD

**Issue by bradleylarrick** [checkstyle/checkstyle#15831]
I'm getting the warning '(extension) LeftCurlyNl: '{' at column 20 should be on a new line'  on a switch statement using lambda-like construct (line 7 below):
```
  for (int i = 0; i < buff.length(); i++) {
    final var ch = buff.charAt(i);
    switch (ch) {
      case 'e' -> buff.setCharAt(i, '3');
      case 'i' -> buff.setCharAt(i, '1');
      case 'o' -> buff.setCharAt(i, '0');
      default -> {
      }
    }
  }
```
Using the latest checkstyle-plugin and checkstyle versions:
```
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-checkstyle-plugin</artifactId>
	<version>3.6.0</version>
	<dependencies>
		<dependency>
			<groupId>com.puppycrawl.tools</groupId>
			<artifactId>checkstyle</artifactId>
			<version>10.19.0</version>
		</dependency>
	</dependencies>
	<configuration>
		<configLocation>google_checks.xml</configLocation>
		<consoleOutput>true</consoleOutput>
		<failsOnError>true</failsOnError>
		<failOnViolation>true</failOnViolation>
		<linkXRef>false</linkXRef>
	</configuration>
</plugin>
```

**Comment by romani** [comment]
@bradleylarrick, please reproduce by CLI it helps us a lot.

@Zopsss , please take a look

**Comment by bradleylarrick** [comment]
Ran
```
java -jar checkstyle-10.19.0-all.jar -c google_checks.xml FooBar.java
```
against this source file:
```
package org.natuna.tools;

/**
 * Test class.
 */
public class FooBar {

  /**
   * Constructor.
   */
  public FooBar() {
  }

  /**
   * Method.
   *
   * @return translated string
   */
  public String getPassword() {
    final var buff = new StringBuilder(32);
    buff.append("Password");

    for (int i = 0; i < buff.length(); i++) {
      final var ch = buff.charAt(i);
      switch (ch) {
        case 'e' -> buff.setCharAt(i, '3');
        case 'i' -> buff.setCharAt(i, '1');
        case 'o' -> buff.setCharAt(i, '0');
        default -> {
        }
      }
    }

    return buff.toString();
  }
}
```
Returned:
```
Starting audit...
[WARN] C:\Users\bradley\temp\FooBar.java:29:20: '{' at column 20 should be on a new line. [LeftCurlyNl]
Audit done.
```

**Comment by bradleylarrick** [comment]
I ran the test class through google-java-format-1.24.0 and got this:
```
     1  package org.natuna.tools;
     2
     3  /** Test class. */
     4  public class FooBarNew {
     5
     6    /** Constructor. */
     7    public FooBarNew() {}
     8
     9    /**
    10     * Method.
    11     *
    12     * @return translated string
    13     */
    14    public String getPassword() {
    15      final var buff = new StringBuilder(32);
    16      buff.append("Password");
    17
    18      for (int i = 0; i < buff.length(); i++) {
    19        final var ch = buff.charAt(i);
    20        switch (ch) {
    21          case 'e' -> buff.setCharAt(i, '3');
    22          case 'i' -> buff.setCharAt(i, '1');
    23          case 'o' -> buff.setCharAt(i, '0');
    24          default -> {}
    25        }
    26      }
    27
    28      return buff.toString();
    29    }
    30  }
```
Running the audit on this new file returned this:
```
Starting audit...
[WARN] C:\Users\bradley\temp\FooBarNew.java:24:20: WhitespaceAround: '{' is not followed by whitespace. Empty blocks                may only be repres
ented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] C:\Users\bradley\temp\FooBarNew.java:24:21: WhitespaceAround: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.
```

**Comment by romani** [comment]
@bradleylarrick, do you agree that we should fix false positives of WhitespaceAround? That we see in checkstyle execution after auto formatting.

**Comment by Zopsss** [comment]
Formatter converted 

```
default -> { 
}
```

into

```
default -> {}
```

this might be a bug in the formatter, as it is not obeying style guide's rule 

https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

```
Exception: In places where these rules allow a single statement ending with a semicolon (;), a block of statements can appear, and the opening brace of this block is preceded by a line break.
```

from our side, the LeftCurlyNl violation was correct.

**Comment by Zopsss** [comment]
> @bradleylarrick, do you agree that we should fix false positives of WhitespaceAround? That we see in checkstyle execution after auto formatting.

We should fix this, this is a false-positive and we were unable to detect it at https://github.com/checkstyle/checkstyle/issues/15338

**Comment by Zopsss** [comment]
We have `allowEmptyLambdas` set to true

https://github.com/checkstyle/checkstyle/blob/ca693c7f33378f483f1740da31634ffac8de6971/src/main/resources/google_checks.xml#L127-L130

but it is still giving error for it.

After modifying `WhitespaceAround`'s suppression and adding `LITERAL_DEFAULT` as following:

```diff
$ diff -u og_suppression.xml modified_suppression.xml
--- og_suppression.xml  2024-10-30 12:03:24.085742000 +0530
+++ modified_suppression.xml    2024-10-30 12:03:38.686326100 +0530
@@ -1,7 +1,7 @@
 <module name="SuppressionXpathSingleFilter">
     <property name="checks" value="WhitespaceAround"/>
     <property name="query" value="//*[self::LITERAL_IF or self::LITERAL_ELSE or self::STATIC_INIT
-                                or self::LITERAL_TRY or self::LITERAL_CATCH]/SLIST[count(./*)=1]
-                                | //*[self::STATIC_INIT or self::LITERAL_TRY or self::LITERAL_IF]
+                                or self::LITERAL_TRY or self::LITERAL_CATCH or LITERAL_DEFAULT]/SLIST[count(./*)=1]
+                                | //*[self::STATIC_INIT or self::LITERAL_TRY or self::LITERAL_IF or LITERAL_DEFAULT]
                                 //*[self::RCURLY][parent::SLIST[count(./*)=1]]"/>
 </module>
```

The false-positive violation got suppressed.

```java
/** Test. */
public class Test {
  /** Test. */
  public static void main(String[] args) {
    switch (ch) {
      default -> {}
    }
  }
}
```

```
$ java -jar .\checkstyle-10.19.0-all.jar -c .\google_checks.xml .\Test.java
Starting audit...
Audit done.
```

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/issues/9540

Looks like we need to activate this new property that we just released

**Comment by romani** [comment]
@bradleylarrick, can you try to change config on your local and run over all your sources to let us catch all use cases.

**Comment by bradleylarrick** [comment]
I ran google-java-format-1.24.0 on another projects, and the ran checkstyle-10.19.0. This code:
```
   310      public String prefix(final Calendar date) {
   311        int year = date.get(Calendar.YEAR);
   312        int month = date.get(Calendar.MONTH) + 1;
   313        int day = date.get(Calendar.DAY_OF_MONTH);
   314
   315        final var claimPrefix = new StringBuilder(pattern.length());
   316        for (int i = 0; i < pattern.length(); i++) {
   317          switch (pattern.charAt(i)) {
   318            case '\\' -> claimPrefix.append(pattern.charAt(++i)); // NOPMD - skipping the escape char
   319            case 'D' -> {
   320              claimPrefix.append((char) ('0' + (day % 10)));
   321              day /= 10;
   322            }
   323            case 'M' -> {
   324              claimPrefix.append((char) ('0' + (month % 10)));
   325              month /= 10;
   326            }
   327            case 'Y' -> {
   328              claimPrefix.append((char) ('0' + (year % 10)));
   329              year /= 10;
   330            }
   331            default -> {}
   332          }
   333        }
   334
   335        // Save the formatted prefix value
   336        currPrefix = claimPrefix.reverse().toString();
   337        return currPrefix;
   338      }
```
generated the following warnings:
```
[WARN] C:\projects\claim-service\src\main\java\org\natuna\claim\service\ClaimNumberFormatter.java:319:23: '{' at column 23 should be on a new line. [L
eftCurlyNl]
[WARN] C:\projects\claim-service\src\main\java\org\natuna\claim\service\ClaimNumberFormatter.java:323:23: '{' at column 23 should be on a new line. [L
eftCurlyNl]
[WARN] C:\projects\claim-service\src\main\java\org\natuna\claim\service\ClaimNumberFormatter.java:327:23: '{' at column 23 should be on a new line. [L
eftCurlyNl]
[WARN] C:\projects\claim-service\src\main\java\org\natuna\claim\service\ClaimNumberFormatter.java:331:22: WhitespaceAround: '{' is not followed by whi
tespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] C:\projects\claim-service\src\main\java\org\natuna\claim\service\ClaimNumberFormatter.java:331:23: WhitespaceAround: '}' is not preceded with w
hitespace. [WhitespaceAround]
```

**Comment by Zopsss** [comment]
Line 319, 323, 327 violation seems correct to me. 

Line 331 violations are false positives, I've provided solution for it in above comment.

**Comment by romani** [comment]
>we can solve it by enabling allowEmptySwitchBlockStatements property

If someone around keyboard and can send PR with fix, I can release it right after merging.

**Comment by Zopsss** [comment]
@romani https://github.com/checkstyle/checkstyle/pull/15839

**Comment by romani** [comment]
Thanks a lot @Zopsss , for quick fix.

@mahfouz72 , your feature is not only life but now in high usage.

@bradleylarrick , fix is released, by same day delivery.
We appreciate it you use it how it works on your code base.

**Comment by bradleylarrick** [comment]
I found a couple more false positives.
````
   123      if (result.length > rpos) {
   124        switch (modulus) {
   125          case 1 -> { // one octet remaining; add two bits and load 2 quintets
   126            workarea = workarea << 2;
   127            rpos = loadCodes(workarea, result, rpos, 2);
   128          }
   129          case 2 -> { // two octet remaining; add four bits and load 4 quintets
   130            workarea = workarea << 4;
   131            rpos = loadCodes(workarea, result, rpos, 4);
   132          }
   133          case 3 -> { // three octets remaining; add one bit and load 5 quintets
   134            workarea = workarea << 1;
   135            rpos = loadCodes(workarea, result, rpos, 5);
   136          }
   137          case 4 -> { // four octets remaining; add 3 bits and load 7 quintets
   138            workarea = workarea << 3;
   139            rpos = loadCodes(workarea, result, rpos, 7);
   140          }
   141          default -> {} // Will never get here
   142        }
   143      }
````
Warnings were:
````
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:125:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:129:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:133:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:137:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
````
And:
````
   246      switch (modulus) {
   247        case 0 -> { // no leftovers
   248        }
   249        case 1, // 5 bits remaining; shouldn't occur from valid encoding, so ignore
   250                2 -> // 10 bits remaining; drop two and output one octet
   251            result[pos++] = (byte) ((workarea >> 2) & MASK_8BITS);
   252        case 3 -> // 15 bits remaining; shouldn't occur from valid encoding
   253            result[pos++] = (byte) ((workarea >> 7) & MASK_8BITS);
   254        case 4 -> // 20 bits remaining; drop the last four and load 2 bytes
   255            pos = loadBytes(workarea >> 4, result, pos, 2);
   256        case 5 -> // 25 bits remaining; drop the last bit and load 3 bytes
   257            pos = loadBytes(workarea >> 1, result, pos, 3);
   258        case 6 -> // 30 bits remaining; shouldn't occur from valid encoding
   259            pos = loadBytes(workarea >> 6, result, pos, 3);
   260        case 7 -> // 35 bits remaining; drop last three and load 4 bytes
   261            pos = loadBytes(workarea >> 3, result, pos, 4);
   262        default -> { // Will never get here
   263        }
   264      }
````
Generated:
````
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:247:17: '{' at column 17 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:262:18: '{' at column 18 should be on a new line. [LeftCurlyNl]
````

**Comment by romani** [comment]
@bradleylarrick, please create new issue.
Please use CLI.
